### ISSUE TITLE
My apps are listed under the wrong namespace

### ISSUE THREAD

**Issue by StarpTech** [k14s/kapp#207]
Hi, I deployed two applications in two different namespaces `staging` and `production`. Why are they listed under "default" in the first column?

```
❯ kapp list -A
Target cluster 'https://xxxx' (nodes: master1, 2+)

Apps in all namespaces

Namespace  Name                 Namespaces  Lcs   Lca  
default    identity				production  true  15h  
^          API				    staging     true  11m
```

Here is another case. It says `Apps in namespace 'default'` but it shows all my apps across all namespaces.

```
❯ kapp list
Target cluster 'https://xxxx' (nodes: master1, 2+)

Apps in namespace 'default'

Name                 Namespaces  Lcs   Lca  
identity  			 production  true  15h  
api				     staging     true  10m  
```

I didn't deploy with `kapp deploy -n $NS` but I specified the namespace in the manifests.

**Environment:**

- kapp version (use `kapp --version`): 0.36.0
- OS (e.g. from `/etc/os-release`): Ubuntu 20.04.2 LTS
- Kubernetes version (use `kubectl version`): v1.19.7

**Comment by gcheadle-vmware** [comment]
Hey @StarpTech,

I'm having trouble re-creating this issue, if you could provide some additional steps for reproducing this scenario, that would be helpful. 

Do you know if the `production` and `staging` namespaces existed prior to your deployment of resources in these namespaces?

> I didn't deploy with kapp deploy -n $NS

Based on the docs, using `-n` flag allows to control which namespace is used for finding and storing metadata ConfigMaps, so specifying the namespace in the manifests, I believe is the way to go.
https://carvel.dev/kapp/docs/latest/state-namespace/

**Comment by StarpTech** [comment]
> Do you know if the production and staging namespaces existed prior to your deployment of resources in these namespaces?

Yes, they existed prior.

> Based on the docs, using -n flag allows to control which namespace is used for finding and storing metadata ConfigMaps, so specifying the namespace in the manifests, I believe is the way to go.

I render my manifests with Skaffold. Kapp for deployment. Just to be clear. My apps are in the corrent namespace only the view is misleading.

**Comment by gcheadle-vmware** [comment]
> My apps are in the correct namespace only the view is misleading.

I agree, I didn't fully understand the output of `kapp list` until today after a conversation with @cppforlife. 

So, for your two kapp apps, `identity` and `API`, they each have a [state config map](https://carvel.dev/kapp/docs/latest/state-namespace/) that kapp creates. This config map tracks metadata about the respective deployed kapp app. You can control where (which namespace) these config maps live in the cluster by providing the `-n` flag, otherwise kapp will use your default. 

In `kapp list -A` command, we're listing all apps in all namespaces, thus I annotated the output with notes in parenthesis :
```
❯ kapp list -A

Apps in all namespaces

Namespace (location of state configmap)    Name        Namespaces (namespaces app resources live)
default                                                               identity     production
^                                                                        API            staging 
```

and the `kapp list` command shows:
```
❯ kapp list

Apps in namespace 'default' (identified by the config maps stored in 'default' NS)

Name                 Namespaces  Lcs   Lca  
identity             production  true  15h  
api	               staging     true  10m  
```

Let me know if this clarifies your understanding. Also, I'm curious how you think this output could be improved?

**Comment by gcheadle-vmware** [comment]
Hey @StarpTech, did my explanation above help?

**Comment by StarpTech** [comment]
Hi @gcheadle-vmware sry, yes it helps but it's still confusing. I'd rename it to `App configs` or remove it.

**Comment by pivotaljohn** [comment]
Tracking the merged request for a more intuitive UX in #217; closing this issue and tracking the work there.
